kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/ConfigWatcherService.java

```diff
@@ -0,0 +1,330 @@
+                    break;
+                }
+
+                if (configFileChanged) {
+                    scheduleConfigurationReload();
+                }
+
+            }
+            catch (InterruptedException e) {
+                LOGGER.warn("Configuration watcher interrupted");
+                Thread.currentThread().interrupt();
+                break;
+            }
+            catch (Exception e) {
+                LOGGER.error("Error in configuration watcher", e);
+                // Continue watching despite errors
+            }
+        }
+    }
+
+    private volatile java.util.concurrent.ScheduledFuture<?> pendingReload;
+
+    private void scheduleConfigurationReload() {
+
+        if (reloadExecutor.isShutdown()) {
+            LOGGER.error("Reload executor is shut down! Cannot schedule configuration reload");
+            return;
+        }
+
+        // Cancel any pending reload to debounce rapid file changes
+        if (pendingReload != null && !pendingReload.isDone()) {
+            pendingReload.cancel(false);
+        }
+
+        // Schedule new reload
+        try {
+            pendingReload = reloadExecutor.schedule(() -> {
+                try {
+                    handleConfigurationChange();
+                }
+                catch (Exception e) {
+                    LOGGER.error("Failed to handle configuration change", e);
+                }
+            }, debounceDelay.toMillis(), TimeUnit.MILLISECONDS);
+        }
+        catch (Exception e) {
+            LOGGER.error("Failed to schedule configuration reload", e);
+        }
+    }
+
+    private void handleConfigurationChange() {
+        try {
+            LOGGER.debug("Processing configuration file change: {}", configFilePath);
+
+            // Check if file exists and is readable
+            if (!Files.exists(configFilePath)) {
+                LOGGER.warn("Configuration file no longer exists: {}", configFilePath);
+                return;
+            }
+
+            // Check if file was actually modified (avoid duplicate reloads)
+            Instant fileModified = Files.getLastModifiedTime(configFilePath).toInstant();
+            LOGGER.debug("File modification time: {}, last processed: {}", fileModified, lastModified);
+
+            if (!fileModified.isAfter(lastModified)) {
+                LOGGER.debug("Configuration file timestamp unchanged - skipping reload");
+                return;
+            }
+
+            LOGGER.info("Configuration file changed, loading new configuration: {}", configFilePath);
+
+            Configuration newConfiguration = loadConfiguration();
+
+            // Only trigger callback if configuration actually changed
+            if (!configurationEquals(currentConfiguration, newConfiguration)) {
+                this.currentConfiguration = newConfiguration;
+                this.lastModified = fileModified;
+
+                try {
+                    onConfigurationChanged.accept(newConfiguration);
```

Comment:
We should have logging here to record that the new configuration is in effect.

But to be honest logging is not really sufficient. We cannot assume there is a human being at a terminal tailing the logs and checking for success or failure. We need something more amenable to the situation where the update is being driven by automation.